File: 26.删除排序数组中的重复项.py
#
# @lc app=leetcode.cn id=26 lang=python
#
# [26] 删除排序数组中的重复项
#

# @lc code=start
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for index in range(len(nums)-1, 0, -1):
            if nums[index] == nums[index-1]:
                nums.pop(index)
        return len(nums)
        
# @lc code=end

File: cpprounttmp/26.删除排序数组中的重复项.cpp
/*
 * @lc app=leetcode.cn id=26 lang=cpp
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        auto unique_end = unique(nums.begin(), nums.end());
        nums.erase(unique_end, nums.end());
        return nums.size();

    }
};
// @lc code=end

